src/coloring.jl

```diff
@@ -529,6 +540,18 @@ function TreeSet(forest::DisjointSets{Tuple{Int,Int}}, nvertices::Int)
                     # (leaf, neighbor) represents the next edge to visit during decompression
                     push!(reverse_bfs_orders[k], (leaf, neighbor))
 
+                    if bool_star
+                        # Initialize the potential hub of the star with the first parent of a leaf
+                        if virtual_hub == 0
+                            virtual_hub = neighbor
+                        else
+                            # Verify if the tree still qualifies as a star
+                            if virtual_hub != neighbor
+                                bool_star = false
+                            end
```

Comment:
I don't understand this logic

Reply:
We define as "virtual hub the first parent of an edge that we encounter.
We then check that the parent of all other edges in the tree is "virtual hub".
If it's true, we have a star.

Reply:
I added two comments in the code. Please don't hesitate to say if it is still unclear.